### ISSUE TITLE
RexTFV updates overwhelm the backend

### ISSUE THREAD

**Issue by dswd** [GLab/ToMaTo#1099]
The web interface issues every 1.2 seconds 5 requests for element updates. Those requests can accumulate if they take longer to execute which can cause an overload situation on the backend.
We might apply an intermediate fix to increase the interval or limit the concurrency but the general problem is that we use polling here. We should instead migrate to a completely different method.

Early idea:
- The hostmanager opens a websocket port that allows the editor to directly communicate with it.
- The backend hands out tokens that the editor uses to communicate with the hostmanager.
- The whole rextfv polling is removed from the backend and the hostmanager only checks the status if requested by a client.

In the long term we could apply this method to all of the communication:
- Apply the observer pattern to all communication instead of polling
- The backend provides tokens to clients that can communicate directly with the hostmanagers
  - If clients change the state of elements, the backend is informed by the host via the observer pattern.
- We might use something like XMPP to have persistent connectivity and for locator/identifier split for elements (which would allow for proxy hosts and for virtual elements provided by some logic on the backend)

**Comment by t-gerhard** [comment]
This means opening a second API on the hostmanager, where authentication tokens are issued by the backend regarding only certain entities. If this is done, we could get rid of the Fileserver, and transfer files between client and hostmanager via this api, too. 

We should allow a socket-based API between backend and editor (eg, socket.io). This would allow the backend to send updates to the editor. In fact, this would enable simultaneous access to a topology on multiple devices. 

I  suggest we talk about this in detail. 

In short term, we could make use of the fact that the backend knows the time when it will next query the rextfv status (this timespan increases). We could add this information to the rextfv status info of elements, and use it to schedule requests in the editor ("next-update time"). 
We could add a collective rextfv status query on topologies to reduce the number of requests. The next-update time would be the minimum of all. 

In the past, I requested a multi-request API call to reduce overhead.

**Comment by t-gerhard** [comment]
I suggest closing this issue and instead opening issues tackling a more general problem regarding status updates.
